conan/cli/commands/graph.py

```diff
@@ -72,11 +72,7 @@ def graph_build_order(conan_api, parser, subparser, *args):
                            help='Reduce the build order, output only those to build. Use this '
                                 'only if the result will not be merged later with other build-order')
     args = parser.parse_args(*args)
-
-    # parameter validation
-    if args.requires and (args.name or args.version or args.user or args.channel):
-        raise ConanException("Can't use --name, --version, --user or --channel arguments with "
-                             "--requires")
+    validate_common_graph_args(args)
```

Comment:
This is a bugfix, we were not validating (because it used to need a --build-requires arg), but making that optional in the validate code gives us back the errors when both defining paths and --requires in this command